# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1711561894 -3600
# Parent  b444bfe46590f7e40c9bc8866baed703f77b1665
Misc: Don't allow stuff inserted by Language Tool to prevent plaintext downgrade.

diff --git a/mailnews/compose/src/nsMsgCompose.cpp b/mailnews/compose/src/nsMsgCompose.cpp
--- a/mailnews/compose/src/nsMsgCompose.cpp
+++ b/mailnews/compose/src/nsMsgCompose.cpp
@@ -4589,8 +4589,20 @@ void nsMsgCompose::TagConvertible(Elemen
   // A style attribute on any element can change layout in any way,
   // so that is not convertible.
   nsAutoString attribValue;
   node->GetAttribute(u"style"_ns, attribValue);
+
+  // Skip stuff inserted by the add-on "Language Tool".
+  if (element.LowerCaseEqualsLiteral("lt-container")) {
+    *_retval = nsIMsgCompConvertible::Plain;
+    return;
+  }
+  if (element.LowerCaseEqualsLiteral("body") &&
+      attribValue.EqualsLiteral("padding-bottom: 1px;")) {
+    *_retval = nsIMsgCompConvertible::Plain;
+    return;
+  }
+
   if (!attribValue.IsEmpty()) {
     *_retval = nsIMsgCompConvertible::No;
     return;
   }
